src/server/mod.rs

```diff
@@ -86,26 +85,52 @@ impl BlockingRequestAction for ShutdownRequest {
     }
 }
 
+pub(crate) fn maybe_notify_unknown_configs<O: Output>(out: &O, unknowns: &Vec<(String)>) {
+    use std::fmt::Write;
+    if unknowns.is_empty() {
+        return;
+    }
+    let mut msg = String::new();
+    for key in unknowns {
+        write!(msg, "unknown RLS configuration: {}; ", key.clone()).ok();
+    }
+    out.notify(Notification::<ShowMessage>::new(ShowMessageParams {
+        typ: MessageType::Warning,
+        message: msg,
+    }));
+}
+
+pub(crate) fn maybe_notify_duplicated_configs<O: Output>(out: &O, dups: &std::collections::HashMap<String, Vec<String>>) {
+    use std::fmt::Write;
+    if dups.is_empty() {
+        return;
+    }
+    let mut msg = String::new();
+    for kv in dups {
+        write!(msg, "{}: ", kv.0).ok();
```

Comment:
Should use `.unwrap()` here, as we know writing to a String will always work. `.ok()` just converts to an `Option`